### ISSUE TITLE
unaltered input->output.

### ISSUE THREAD

**Issue by gacord23** [lovell/sharp#4245]
### What are you trying to achieve?
I would like to take an image from input, to output as either the original input type, or raw (this would include extractChannel->raw) without any icc transforms or colorspace alterations.  I've had success on my own fork by adding an {icc: 'noop'} option to withMetadata which works fine with RGB images.  But if I have a CMYK image (my CMYK images may, or may not, have an embedded profile), I'm seeing it still convert it to an RGB colorspace and I really want zero transforming of the pixels through the pipeline.

Even if it requires modifying my own fork further, Any insight on how I might achieve this?

**Comment by lovell** [comment]
Something like the following might get you closer to what you need, but most software requires CMYK images to contain a profile as they are device-dependent.
```js
const cmykData = await sharp(cmykInput, { ignoreIcc: true })
  .pipelineColourspace('cmyk')
  .toColourspace('cmyk')
  .raw()
  .toBuffer();
```

https://sharp.pixelplumbing.com/api-constructor
https://sharp.pixelplumbing.com/api-colour#pipelinecolourspace
https://sharp.pixelplumbing.com/api-colour#tocolourspace

**Comment by gacord23** [comment]
Right, I have a particular need for processing them both with and without profiles and do all of my color workflow via LCMS.  So, would sharp completely bypass any transform in this case or would it still attempt to do something like move it to some rgb16 and back again?

**Comment by lovell** [comment]
@gacord23 Did you try the sample code I provided? Were you able to make any progress with this?

**Comment by lovell** [comment]
Closing due to inactivity but please feel free to reopen with more details if further help is required.
